File: src/main/java/mods/orca/mffs/items/ItemBase.java
package mods.orca.mffs.items;

import mods.orca.mffs.client.MFFSTab;
import net.minecraft.item.Item;

public class ItemBase extends Item {

    protected String name;

    public ItemBase(String name, Boolean setCreativeTab) {
        this.name = name;

        setTranslationKey("mffs." + name);
        setRegistryName(name);

        if (setCreativeTab) setCreativeTab(MFFSTab.mffsTab);

        ModItems.ITEMS.add(this);
    }

    public ItemBase(String name) {
        this(name, true);
    }
}

File: src/main/java/mods/orca/mffs/items/ItemBase.kt
package mods.orca.mffs.items

import mods.orca.mffs.client.MFFSTab
import net.minecraft.item.Item

open class ItemBase @JvmOverloads constructor(
    protected var name: String,
    setCreativeTab: Boolean = true
) : Item() {

    private fun addToCreativeTab() {
        setCreativeTab(MFFSTab)
    }

    private fun register() {
        ModItems.ITEMS.add(this)
    }

    init {

        translationKey = "mffs.$name"
        setRegistryName(name)

        if (setCreativeTab) {
            addToCreativeTab()
        }

        register()

    }

}
